### ISSUE TITLE
joint name of each 3d coordinates

### ISSUE THREAD

**Issue by TheGameIsAboutGlory** [facebookresearch/VideoPose3D#57]
First of all, Thanks for your wonderful works

I choose random youtube videos
and I extract 3d coordinates of every 17joints of every Frames in Video
Because I got the total 253 frames, I finally got [253, 17, 3]format of numpy.ndarray

But the thing is... I want to know which one matches to which joint.... like this one is for left knee just like this....

Sorry for Rudimentary Question.

If there is any part of code lines that will helpful to understand about this question,
Please let me know

Thanks!

**Comment by zh-plus** [comment]
You can refer to https://github.com/MVIG-SJTU/AlphaPose/blob/master/doc/output.md#keypoint-ordering.
And 17 points indicate the default keypoint order fo COCO.

Hope this helps!

**Comment by dariopavllo** [comment]
The answer by @zh-plus indicates the keypoint names for the input layout (COCO).

If you also want to know the output layout (Human3.6M), it should be as follows:

['Hip (root)', 'Right hip', 'Right knee', 'Right foot', 'Left hip', 'Left knee', 'Left foot', 'Spine', 'Thorax', 'Nose', 'Head', 'Left shoulder', 'Left elbow', 'Left wrist', 'Right shoulder', 'Right elbow', 'Right wrist']

I might have inverted L and R, so please double check.

**Comment by TheGameIsAboutGlory** [comment]
> The answer by @zh-plus indicates the keypoint names for the input layout (COCO).
> 
> If you also want to know the output layout (Human3.6M), it should be as follows:
> 
> ['Hip (root)', 'Right hip', 'Right knee', 'Right foot', 'Left hip', 'Left knee', 'Left foot', 'Spine', 'Thorax', 'Nose', 'Head', 'Left shoulder', 'Left elbow', 'Left wrist', 'Right shoulder', 'Right elbow', 'Right wrist']
> 
> I might have inverted L and R, so please double check.

Thanks For your reply!
Actually i'm using 'https://github.com/tobiascz/VideoPose3D' this fork version.
By using this i can get 2d keypoints like below for each frames.
<img width="163" alt="213" src="https://user-images.githubusercontent.com/39264008/61281579-d2ae2e00-a7f4-11e9-8e7d-1b2dc9923adb.png">

And for now, I guess @zh-plus's reply is right for 2d skeleton detections cuz i find the code like below.
<img width="373" alt="4" src="https://user-images.githubusercontent.com/39264008/61281688-08ebad80-a7f5-11e9-9b71-72c6670980ca.png">

But i wonder whether the values i got from 3d skeleton detections is the same as 2d keypoints.
i mean.. 
<img width="245" alt="5" src="https://user-images.githubusercontent.com/39264008/61281829-43554a80-a7f5-11e9-945d-0e804096c3ae.png">
And That value is like below
PREDICTION PRINT : [[[ 5.54544204e-06 -2.21030677e-05 -6.80828180e-06]
  [-1.06989525e-01 -7.34037254e-03  8.69212598e-02]
  [-1.39160663e-01  4.32419509e-01  2.84093976e-01]
  ...
  [-1.32404387e-01 -4.24458891e-01 -5.67590073e-03]
  [-1.82774305e-01 -1.67134076e-01  1.34978592e-01]
  [-2.05606341e-01  7.87530318e-02  1.08486563e-01]]

 [[ 5.51107587e-06 -2.20942256e-05 -6.81388656e-06]
  [-1.06980890e-01 -7.30840489e-03  8.68559703e-02]
  [-1.39163867e-01  4.32386756e-01  2.84141719e-01]
  ...
  [-1.32346094e-01 -4.24305558e-01 -5.78352623e-03]
  [-1.82729900e-01 -1.66985691e-01  1.34543389e-01]
  [-2.05554113e-01  7.87367672e-02  1.07708186e-01]]

 [[ 5.51074663e-06 -2.20796464e-05 -6.81893061e-06]
  [-1.06980838e-01 -7.31971208e-03  8.68046731e-02]
  [-1.39065012e-01  4.32274997e-01  2.84012765e-01]
  ...
  [-1.32142290e-01 -4.24136400e-01 -5.75241446e-03]
  [-1.82443306e-01 -1.66848943e-01  1.34416938e-01]
  [-2.05413997e-01  7.87053257e-02  1.07242785e-01]]

 ...

 [[ 4.82084124e-06 -2.45743413e-05 -6.86694466e-06]
  [-1.09334119e-01 -8.05263408e-03  8.01904202e-02]
  [-1.89210966e-01  4.29886520e-01  2.64010608e-01]
  ...
  [-1.25356734e-01 -4.28942561e-01 -1.90622862e-02]
  [-1.76709875e-01 -1.70224845e-01  1.09223455e-01]
  [-2.28960335e-01  5.35771251e-02  9.05488580e-02]]

 [[ 4.15030354e-06 -2.42134665e-05 -6.97244741e-06]
  [-1.11068733e-01 -8.23098794e-03  7.83343315e-02]
  [-1.67399436e-01  4.31013167e-01  2.67257065e-01]
  ...
  [-1.23177543e-01 -4.30679977e-01 -1.65667981e-02]
  [-1.76823229e-01 -1.71831280e-01  1.04506500e-01]
  [-2.27739602e-01  5.44780530e-02  7.67418668e-02]]

 [[ 3.46747402e-06 -2.38618522e-05 -7.01604677e-06]
  [-1.11583829e-01 -7.66054541e-03  7.73284584e-02]
  [-1.47479475e-01  4.30575967e-01  2.72444189e-01]
  ...
  [-1.20109528e-01 -4.32742625e-01 -1.52162593e-02]
  [-1.78853214e-01 -1.76297665e-01  1.04192711e-01]
  [-2.27402717e-01  5.27202412e-02  7.48873055e-02]]]

I got the 3d values like i attached up.
i'm curious if the fitst '[ 5.54544204e-06 -2.21030677e-05 -6.80828180e-06]' this value is for the nose's
x,y,z coordinate value

**Comment by dariopavllo** [comment]
No, they are not the same. Inputs and outputs use different layouts. The inputs are in COCO layout, the outputs are in Human3.6M. They have the same number of joints (17), but some of them are different or are in different order.

See my previous answer for the list Human3.6M joints.

**Comment by TheGameIsAboutGlory** [comment]
> No, they are not the same. Inputs and outputs use different layouts. The inputs are in COCO layout, the outputs are in Human3.6M. They have the same number of joints (17), but some of them are different or are in different order.
> 
> See my previous answer for the list Human3.6M joints.

wow!! 
So.. you mean [17, 3] numpy array means 
['Hip (root)', 'Right hip', 'Right knee', 'Right foot', 'Left hip', 'Left knee', 'Left foot', 'Spine', 'Thorax', 'Nose', 'Head', 'Left shoulder', 'Left elbow', 'Left wrist', 'Right shoulder', 'Right elbow', 'Right wrist']'s x,y,z coordinates? sequentially?

**Comment by dariopavllo** [comment]
Yes

**Comment by kazukiotsuka** [comment]
```python
class Keypoints3D(Enum):
    HIP = 0
    R_HIP = 1
    R_KNEE = 2
    R_FOOT = 3
    L_HIP = 4
    L_KNEE = 5
    L_FOOT = 6
    SPINE = 7
    THORAX = 8
    NOSE = 9
    HEAD = 10
    L_SHOULDER = 11
    L_ELBOW = 12
    L_WRIST = 13
    R_SHOULDER = 14
    R_ELBOW = 15
    R_WRIST = 16
```
this works for me.

**Comment by TheGameIsAboutGlory** [comment]
> ```python
> class Keypoints3D(Enum):
>     HIP = 0
>     R_HIP = 1
>     R_KNEE = 2
>     R_FOOT = 3
>     L_HIP = 4
>     L_KNEE = 5
>     L_FOOT = 6
>     SPINE = 7
>     THORAX = 8
>     NOSE = 9
>     HEAD = 10
>     L_SHOULDER = 11
>     L_ELBOW = 12
>     L_WRIST = 13
>     R_SHOULDER = 14
>     R_ELBOW = 15
>     R_WRIST = 16
> ```
> 
> this works for me.

Thanks for reply!!
can you tell me where 'class Keypoints3D(Enum) : ' this code exist?

**Comment by Justinemmerich** [comment]
guys could you give me a hint how to access the array i got a shape of for example (24,17,3) where 17,3 is one set of keypoints and 24 is the number of frames processed. np[0] gives me all 3D keypoints of the Hip in all frames.  How can I access the first frame and all 17 Keypoints ? thanks al lot

**Comment by shox21512** [comment]
Is it possible to add the name of joint to 3D joint coordinate points?
If so, how we can do that?
